ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx

```diff
@@ -27,12 +84,277 @@ const translations = defineMessages({
   }
 });
 
-export default function AudiencesPanel() {
+interface AudiencesPanelUIProps {
+  audiencesResource: any;
+  onFormChange: Function;
+  saveProfile: Function;
+  setDefaults: Function;
+}
+
+export function AudiencesPanelUI(props: AudiencesPanelUIProps) {
+
+  const classes = useStyles({});
+  const {
+    audiencesResource,
+    onFormChange,
+    saveProfile,
+    setDefaults
+  } = props;
+  const audiencesData = audiencesResource.read();
+  const { config, profile } = audiencesData;
+
   return (
     <ToolPanel title={translations.audiencesPanel}>
-      <Typography component="h2" variant="subtitle1" style={{ padding: '10px' }}>
-        Audiences Panel
-      </Typography>
+      {
+        <>
+          <Grid className={classes.PanelMargin}>
+            {
+              Object.keys(config.fields).map((property: any, index: number) => (
+                <div key={index}>
+                  <AudiencesFormSection
+                    property={config.fields[property]}
+                    profileValue={profile[property] ? profile[property].key : undefined}
+                    profileTimezone={profile[`${property}_tz`] ? profile[`${property}_tz`].key : undefined}
```

Comment:
This field awkward.

Could be something like:

```js
...map(field => {
  const Control = controlsMap[field.type];
  return (
    <AudiencesFormSection field={field}>
      <Control field={field} value={profile[field.id]} onChange={onFieldChange} />
    </AudiencesFormSection>
  );
})
```

AudiencesFormSection renders the base structure, label, helpText and the control, well, the control.